filebeat/input/container/input.go

```diff
@@ -0,0 +1,117 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package container
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/elastic/beats/filebeat/channel"
+	"github.com/elastic/beats/filebeat/input"
+	"github.com/elastic/beats/filebeat/input/log"
+	"github.com/elastic/beats/libbeat/common"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	err := input.Register("container", NewInput)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// NewInput creates a new container input
+func NewInput(
+	cfg *common.Config,
+	outletFactory channel.Connector,
+	context input.Context,
+) (input.Input, error) {
+	// Wrap log input with custom docker settings
+	config := defaultConfig
+	if err := cfg.Unpack(&config); err != nil {
+		return nil, errors.Wrap(err, "reading container input config")
+	}
+
+	if err := validateStream(config.Stream); err != nil {
+		return nil, err
+	}
+
+	if err := validateFormat(config.Format); err != nil {
+		return nil, err
+	}
```

Comment:
Better implement `func (c *config) Validate() error `. This will trigger validation on Unpack and adds some more context (file, full name of parent namespace including index) to the error message.